src/System.CommandLine.Tests/Binding/TypeConversionTests.cs

```diff
@@ -493,16 +493,18 @@ public void A_default_value_with_a_custom_constructor_can_be_specified_for_a_com
         {
             var directoryInfo = new DirectoryInfo(Directory.GetCurrentDirectory());
 
+            var argument = new Argument<DirectoryInfo>("the-arg", () => directoryInfo);
+
             var command = new Command("something")
             {
-                new Argument<DirectoryInfo>("the-arg", () => directoryInfo)
+                argument
             };
 
             var result = command.Parse("something");
 
             result.Errors.Should().BeEmpty();
 
-            var value = result.CommandResult.GetArgumentValueOrDefault("the-arg");
+            var value = result.ValueForArgument("the-arg");
```

Comment:
That's a strange method name. Method names should start with verbs_ `GetValueForArgument`

I guess it's like `Newtonsoft.Json`. A parameterized property.

Reply:
We'll likely rename these: #1127.